pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -289,9 +289,12 @@ func (iqr *IQR) readAllColumnsWithRRCs() (map[string][]utils.CValueEnclosure, er
 		}
 	}
 
-	for oldName := range iqr.renamedColumns {
-		// TODO: don't read these columns from the RRCs, instead of reading and
-		// then deleting them.
+	for oldName, newName := range iqr.renamedColumns {
+		_, exists := results[oldName]
+		if !exists {
+			continue
+		}
+		results[newName] = results[oldName]
 		delete(results, oldName)
 	}
```

Comment:
Do we need this? I see that now on line 272 we're not reading the `deletedColumns` from RRCs. I thought that when we made a call to rename columns we'd already do this, and so doing this here would be a no-op. Am I missing something?

I'm also concerned: I don't think we need this here because we should do it elsewhere, but if we keep this change because it does actually have an effect in some cases, it's going to get harder and harder to remove this code

Reply:
We do not mark them as `deletedColumns`, cause they are not deleted, they are just renamed.